x-pack/plugins/cloud_security_posture/public/pages/rules/index.tsx

```diff
@@ -79,21 +80,21 @@ export const Rules = ({ match: { params } }: RouteComponentProps<PageUrlParams>)
               defaultMessage="{integrationType}, {agentPolicyName}"
               values={{
                 integrationType: integrationInfo.data.package.title,
-                agentPolicyName: integrationInfo.data.name,
+                agentPolicyName: integrationAgentInfo?.data?.name,
               }}
             />
           </EuiTextColor>
         ),
       },
     }),
-    [http.basePath, integrationInfo.data, params]
+    [http.basePath, integrationAgentInfo?.data?.name, integrationInfo.data, params]
   );
 
   return (
     <>
       <CspPageTemplate
         {...pageProps}
-        query={integrationInfo}
+        query={[integrationInfo, integrationAgentInfo]}
```

Comment:
the current behaviour looks at the query we passed to decide what state to show - loading/error/success

but some pages may need more queries. (example: findings page)

to make this nicer to use, we can now pass multiple queries.